projects/ccd-case-ui-toolkit/src/lib/shared/components/palette/query-management/components/query-details/query-details.component.ts

```diff
@@ -1,18 +1,29 @@
-import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { Component, EventEmitter, Input, OnChanges, Output } from '@angular/core';
 import { SessionStorageService } from '../../../../../services';
 import { QueryListItem } from '../../models';
-
+import { ActivatedRoute, Router } from '@angular/router';
+import { Constants } from '../../../../../commons/constants';
 @Component({
   selector: 'ccd-query-details',
   templateUrl: './query-details.component.html',
   styleUrls: ['./query-details.component.scss']
 })
-export class QueryDetailsComponent {
+export class QueryDetailsComponent implements OnChanges{
   @Input() public query: QueryListItem;
   @Output() public backClicked: EventEmitter<boolean> = new EventEmitter();
   @Input() public caseId: string;
+  @Output() public hasResponded: EventEmitter<boolean> = new EventEmitter();
+  @Input() public totalNumberOfQueryChildren: number;
+
+  public showLink: boolean = true;
+  private static readonly QUERY_ITEM_RESPOND = '3';
+  private queryItemId: string;
+  public message: string;
```

Comment:
Minor pedantic comment - Obvs was not sorted before but would prefer Input() and Output() grouped with each other, same with public fields